runtime/interpreter/value_storagecapabilitycontroller.go

```diff
@@ -19,115 +19,212 @@
 package interpreter
 
 import (
-	"fmt"
+	"github.com/onflow/atree"
 
 	"github.com/onflow/cadence/runtime/common"
 	"github.com/onflow/cadence/runtime/errors"
 	"github.com/onflow/cadence/runtime/format"
 	"github.com/onflow/cadence/runtime/sema"
 )
 
-var storageCapabilityControllerFieldNames = []string{
-	sema.StorageCapabilityControllerTypeBorrowTypeFieldName,
-	sema.StorageCapabilityControllerTypeCapabilityIDFieldName,
+type CapabilityControllerValue interface {
+	Value
+	isCapabilityControllerValue()
+}
+
+// StorageCapabilityControllerValue
+
+type StorageCapabilityControllerValue struct {
+	BorrowType       StaticType
+	TargetPath       PathValue
+	CapabilityID     UInt64Value
+	RetargetFunction FunctionValue
```

Comment:
How does the `RetargetFunction` part of this get stored? Or are we not storing the function?